python/graphstorm/dataloading/dataset.py

```diff
@@ -76,13 +75,27 @@ def prepare_batch_input(g, input_nodes,
             # concatenate multiple features together
             feats = []
             for fname in feat_name:
+                assert fname in g.nodes[ntype].data, \
+                    f"{fname} does not exist as a node feature of {ntype}"
                 data = g.nodes[ntype].data[fname]
                 if is_wholegraph_embedding(data):
                     data = data.gather(nid.to(dev))
                 else:
                     data = data[nid].to(dev)
                 feats.append(data)
-            feat[ntype] = th.cat(feats, dim=1)
+            assert len(feats) > 0, \
+                "No feature exists in the graph. " \
+                f"Expecting the graph have following node features {feat_name}."
+
+            if len(feats[0].shape) == 1:
```

Comment:
For 1D features, it should be stored as `(num_nodes, 1)` instead of `(num_nodes,)`.
GConstruct and GSProcessing will automatically convert 1D features into 2D like https://github.com/awslabs/graphstorm/blob/main/python/graphstorm/gconstruct/transform.py#L1310-L1316

Reply:
There is one case where users can still use DGL graph as the input of graph partition. So we might not be able to control the dimension of features. In such case, it could be better to change this `if len...` to an assertion to let users know they must store feature in `(num_nodes, 1)` format.

Reply:
Update: https://graphstorm.readthedocs.io/en/latest/tutorials/own-data.html#option-2-required-dgl-graph

Reply:
All nodes/edges features must be 2D tensors.

Reply:
Updated